src/VisualStudio/IntegrationTest/TestUtilities/InProcess/SolutionExplorer_InProc.cs

```diff
@@ -296,37 +298,86 @@ public int GetErrorListErrorCount()
             return errorCount;
         }
 
-        public void OpenFile(string projectName, string relativeFilePath)
+        public void OpenFileWithDesigner(string projectName, string relativeFilePath)
         {
-            EnvDTE.Project project = null;
-            for (int i = 1; i <= _solution.Projects.Count; i++)
-            {
-                if (_solution.Projects.Item(i).Name == projectName)
-                {
-                    project = _solution.Projects.Item(i);
-                    break;
-                }
-            }
+            var filePath = GetFilePath(projectName, relativeFilePath);
+            var fileName = Path.GetFileName(filePath);
+            var project = _solution.Projects.Cast<EnvDTE.Project>().First(x => x.Name == projectName);
+            var window = project.ProjectItems.Item(fileName).Open(EnvDTE.Constants.vsViewKindDesigner);
+            window.Activate();
 
-            if (project == null)
+            var dte = GetDTE();
+            while (!dte.ActiveWindow.Caption.Contains(fileName))
             {
-                throw new InvalidOperationException($"Could not find project '{projectName}'.");
+                Thread.Yield();
             }
+        }
 
-            var projectPath = Path.GetDirectoryName(project.FullName);
+        public void OpenFile(string projectName, string relativeFilePath)
+        {
+            var filePath = GetFilePath(projectName, relativeFilePath);
+            var fileName = Path.GetFileName(filePath);
 
-            var filePath = Path.Combine(projectPath, relativeFilePath);
             ExecuteCommand("File.OpenFile", filePath);
 
             var dte = GetDTE();
-            var fileName = Path.GetFileName(filePath);
-
             while (!dte.ActiveWindow.Caption.Contains(fileName))
             {
                 Thread.Yield();
             }
         }
 
+        public void CloseFile(string projectName, string relativeFilePath, bool saveFile)
+        {
+            var filePath = GetFilePath(projectName, relativeFilePath);
+            var fileName = Path.GetFileName(filePath);
+
+            var dte = GetDTE();
+            var fileToClose = dte.Documents.Cast<EnvDTE.Document>().FirstOrDefault(document => document.Name.Equals(fileName));
```

Comment:
You're using `FirstOrDefault` but not checking for the default. Either add the check, or just use `First` (or perhaps `Single` for a stronger guarantee).